ee/clickhouse/queries/funnels/funnel_time_to_convert.py

```diff
@@ -60,10 +60,19 @@ def get_query(self) -> str:
         ]
         steps_average_conversion_time_expression_sum = " + ".join(steps_average_conversion_time_identifiers)
 
+        steps_average_conditional_for_invalid_values = [
+            f"{identifier} >= 0" for identifier in steps_average_conversion_time_identifiers
+        ]
+        # this is protection against the CH bug: https://github.com/ClickHouse/ClickHouse/issues/26580
+        #  once the issue is resolved, stop skipping the test: test_auto_bin_count_single_step_duplicate_events
+        # and remove this comment
+
         query = f"""
             WITH
                 step_runs AS (
-                    {steps_per_person_query}
+                    SELECT * FROM (
+                        {steps_per_person_query}
```

Comment:
Q: Why not push the predicates in here?

Reply:
> We aren't doing this at a lower level (in, say, the original funnel class) because the other things work fine, and the following safeguards in those places (increasing ordering of event times) ensure we don't run into this issue.

Good question! I see good arguments for either, but opted for "where the problem shows up", as it also keeps things in one place. Doing it in `steps_per_person_query` would also mean doing it in _all_ types of funnel orderings.